### PR TITLE
add SQL test case to moudle sql-test and parse-test

### PR DESCRIPTION
**Pull request by haetao**
Add test case
Case1: insert_with_duplicate_key_way_1
Case2: insert_with_duplicate_key_way_2
Case3: insert_with_duplicate_key_way_3

### CODE REVIEW

File: sharding-core/sharding-core-parse/sharding-core-parse-test/src/test/resources/sharding/dml/insert.xml [hunk 1]
```diff
@@ -665,4 +665,79 @@
             </and-condition>
         </sharding-conditions>
     </parser-result>
+
+    <parser-result sql-case-id="insert_with_duplicate_key_way_1" parameters="1, 2, 3">
```
Comment (terrymanu) [inline sharding-core/sharding-core-parse/sharding-core-parse-test/src/test/resources/sharding/dml/insert.xml]: Do not name as `xxx_1` and `xxx_2`, please name the test case make sense.

Comment (haetao) [reply sharding-core/sharding-core-parse/sharding-core-parse-test/src/test/resources/sharding/dml/insert.xml]: I 'll  modify them.

File: sharding-sql-test/src/main/resources/sql/sharding/dml/insert.xml [hunk 2]
```diff
@@ -48,4 +48,7 @@
     <sql-case id="insert_with_null_value" value="INSERT INTO t_null_value_test(col1) VALUES(null)" db-types="MySQL" />
     <sql-case id="insert_with_blob_value" value="INSERT INTO t_blob_value_test(col1) VALUES(_BINARY'This is a binary value.')" db-types="MySQL" />
     <sql-case id="insert_with_function" value="INSERT INTO t_order(current_date, order_id, user_id) VALUES (curdate(), ?, ?)" db-types="MySQL" />
+    <sql-case id="insert_duplicate_key_update_with_equation_common" value="INSERT INTO emp(id,age,salary) VALUES (?,?,?) ON DUPLICATE KEY UPDATE salary = salary+1" db-types="MySQL"/>
+    <sql-case id="insert_duplicate_key_update_two_group_data" value="INSERT INTO emp(id,age,salary) VALUES (?,?,?),(?,?,?) ON DUPLICATE KEY UPDATE salary = VALUES(id)+VALUES(age)" db-types="MySQL"/>
+    <sql-case id="insert_duplicate_key_update_with_equation_values" value="INSERT INTO emp(id,age,salary) VALUES (?,?,?) ON DUPLICATE KEY UPDATE salary = VALUES(id)" db-types="MySQL"/>
```
Comment (terrymanu) [inline sharding-sql-test/src/main/resources/sql/sharding/dml/insert.xml]: What's mean of equation values? maybe values function is better?

Comment (haetao) [reply sharding-sql-test/src/main/resources/sql/sharding/dml/insert.xml]: equation means  "a = v1+v2".   v1 or v2 may be a value or a function .

Comment (haetao) [reply sharding-sql-test/src/main/resources/sql/sharding/dml/insert.xml]: I 'll fix it tomorrow.

**Comment by haetao** [comment]
Adjust "equation" to "function". Function is better than Equation.

File: sharding-sql-test/src/main/resources/sql/sharding/dml/insert.xml [hunk 3]
```diff
@@ -48,4 +48,7 @@
     <sql-case id="insert_with_null_value" value="INSERT INTO t_null_value_test(col1) VALUES(null)" db-types="MySQL" />
     <sql-case id="insert_with_blob_value" value="INSERT INTO t_blob_value_test(col1) VALUES(_BINARY'This is a binary value.')" db-types="MySQL" />
     <sql-case id="insert_with_function" value="INSERT INTO t_order(current_date, order_id, user_id) VALUES (curdate(), ?, ?)" db-types="MySQL" />
+    <sql-case id="insert_duplicate_key_update_with_function_common" value="INSERT INTO emp(id,age,salary) VALUES (?,?,?) ON DUPLICATE KEY UPDATE salary = salary+1" db-types="MySQL"/>
```
Comment (terrymanu) [inline sharding-sql-test/src/main/resources/sql/sharding/dml/insert.xml]: what's mean of `with_function_common`?

File: sharding-sql-test/src/main/resources/sql/sharding/dml/insert.xml [hunk 4]
```diff
@@ -48,4 +48,7 @@
...
+    <sql-case id="insert_duplicate_key_update_two_group_data" value="INSERT INTO emp(id,age,salary) VALUES (?,?,?),(?,?,?) ON DUPLICATE KEY UPDATE salary = VALUES(id)+VALUES(age)" db-types="MySQL"/>
```
Comment (terrymanu) [inline sharding-sql-test/src/main/resources/sql/sharding/dml/insert.xml]: I think `with_function_values` should be `with_values_function`

Comment (terrymanu) [inline sharding-sql-test/src/main/resources/sql/sharding/dml/insert.xml]: What's mean `two_group_data`? We may name test cases more accurate.
Perhaps `with_complicated_expression` is a better name.

File: sharding-sql-test/src/main/resources/sql/sharding/dml/insert.xml [hunk 5]
```diff
@@ -48,4 +48,7 @@
...
+    <sql-case id="insert_duplicate_key_update_with_function_values" value="INSERT INTO emp(id,age,salary) VALUES (?,?,?) ON DUPLICATE KEY UPDATE salary = VALUES(id)" db-types="MySQL"/>
```
Comment (terrymanu) [inline sharding-sql-test/src/main/resources/sql/sharding/dml/insert.xml]: The SQL fragment of `salary = VALUES(id)` does not make sense, use `id = VALUES(id)` is better

File: sharding-sql-test/src/main/resources/sql/sharding/dml/insert.xml (hunk 4 above)
Comment (terrymanu) [inline sharding-sql-test/src/main/resources/sql/sharding/dml/insert.xml]: The SQL fragment of `salary = VALUES(id)+VALUES(age)` does not make sense, use `salary = VALUES(salary)+VALUES(salary) + 10` is better

File: sharding-sql-test/src/main/resources/sql/sharding/dml/insert.xml (hunk 3 above)
Comment (terrymanu) [inline sharding-sql-test/src/main/resources/sql/sharding/dml/insert.xml]: I don't think this case is different with the case `insert_duplicate_key_update_two_group_data`.
Can we merge the 2 cases? What I mean is can we just remove this case?

**Comment by haetao** [comment]
I fixed theses detail and abridged SQL case.

**Pull request merged**